src/DynamoCore/Core/Threading/DelegateBasedAsyncTask.cs

```diff
@@ -23,7 +23,7 @@ internal DelegateBasedAsyncTask(IScheduler scheduler)
         {
         }
         
-        internal DelegateBasedAsyncTask(IScheduler scheduler, Action action)
+        public DelegateBasedAsyncTask(IScheduler scheduler, Action action)
```

Comment:
By design `DelegateBasedAsyncTask` is meant for external consumption anyway, so this looks pretty good to me.